### PR TITLE
[RHPAM-2634] Remote Server is lot from the controller after Kie Server scale down and scale up

### PR DESCRIPTION
**Pull request by rhtevan**
**_Issue Description_**
Given a kie controller and kie server deployed in the environment, and some containers deployed, when I scale down the kie server and then I scale it up again, then the kie controller loses the remote server (kie server url).

Related JIRA link:
https://issues.redhat.com/projects/RHPAM/issues/RHPAM-2634

Related PRs
https://github.com/jboss-container-images/jboss-kie-modules/pull/346
https://github.com/jboss-container-images/rhpam-7-openshift-image/pull/411
https://github.com/kiegroup/kie-cloud-operator/pull/343

Signed-off-by: Evan Zhang <evan.zhang@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [x] Pull Request title is properly formatted: `[KIECLOUD-XYZ] Subject`, `[RHDM-XYZ] Subject` or `[RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted
- [x] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [x] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by rhtevan** [comment]
@spolti  @sutaakar Could you help take a look. Thanks!

**Comment by rhtevan** [comment]
@sutaakar Will need some regression tests for https://issues.redhat.com/browse/RHPAM-2008 as well. Thanks!

File: jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh [hunk 1]
```diff
@@ -56,8 +56,7 @@ if [ -n ${WORKBENCH_SERVICE_NAME} -a -n "${KIE_SERVER_ID}" ]; then
         fi
 
     elif [ ${kieReplicas} -gt 0 ]; then
-        log_info "KIE Server Replicas is ${kieReplicas}, updating ${KIE_SERVER_ID} configMap to USED."
-        update_config_map "{\"metadata\":{\"labels\":{\"services.server.kie.org/kie-server-state\":\"USED\"}}}" "${kieDCName}"
+        log_info "KIE Server Replicas is ${kieReplicas}, connecting ${KIE_SERVER_ID} to BC/WB if exists."
```
Comment (spolti) [inline jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh]: Wouldn't it be necessary anymore?

Comment (rhtevan) [reply jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh]: Keeping the info log which contains KieServer Id and number of replicas is mainly for better debug-ability.
The analysis for not updating CM label has been elaborated [here](https://issues.redhat.com/browse/RHPAM-2634?focusedCommentId=13947321&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13947321). It is unlikely to bring any negative side effects.

**Comment by sutaakar** [comment]
Same behaviour is for 7.4

**Comment by rhtevan** [comment]
@spolti @errantepiphany @sutaakar 
Could you help take a look, not sure why CI was failing.

**Comment by sutaakar** [comment]
@rhtevan Just tried it on OCP 4.3 with latest images, the issue with popup error is still there even when this change is applied.

**Comment by rhtevan** [comment]
@sutaakar  The event sequence required is a bit tricky between scale-up and scale-down scenarios. Two key constraints: 1. state label change MUST happen before notifying BC; 2. Before notifying BC, the ServerTemplate cache MUST expire. Tested the latest tweak with 7.6.0 images. It works fine now.  Appreciated very much if you could give another try. (Please change the default ServerTemplate cache TTL to lower than 10 seconds, i.e. 5 seconds)

@spolti  @errantepiphany  Could you help review it, appreciated!

**Comment by rhtevan** [comment]
@jakubschwan Could help check if this fix addresses the RHPAM-2634 issue. Thanks!

**Comment by Sgitario** [comment]
@rhtevan the latest fix works for me.

**Comment by spolti** [comment]
@sutaakar do you mind to take a look before we merge this?

**Comment by sutaakar** [comment]
@spolti I would like to check it first against latest master artifacts (built today)
Can you please adjust images to use latest binaries?

**Comment by spolti** [comment]
@sutaakar just updated, the latest build is 20200129.

**Comment by spolti** [comment]
@sutaakar were you able to test it?

**Comment by sutaakar** [comment]
@spolti I think that the problem is related with https://github.com/kiegroup/droolsjbpm-integration/pull/1995 , which is still opened.
@rhtevan Is it so?

**Comment by rhtevan** [comment]
@sutaakar It would be better to treat these two issues separately. This PR fixes the missing 'org.kie.server.location' issue specified at RHPAM-2634. During testing, a new issue has been identified and logged by [RHPAM-2657](https://issues.redhat.com/projects/RHPAM/issues/RHPAM-2657), which is related to some recent code changes brought by 7.7 code base. At the moment, the fix can be verified with 7.6 image. We could get this PR merge first. Wdyt?

**Pull request merged**